### ISSUE TITLE
Remote development with WSL toolchain selected results in wrong command being run

### ISSUE THREAD

**Issue by mellowagain** [intellij-rust/intellij-rust#9018]
## Environment

* **IntelliJ Rust plugin version:** 0.4.173.4714-221
* **Rust toolchain version:** 1.61.0 (fe5b13d68 2022-05-18) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2022.1.3 (CL-221.5921.27)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

If one has the WSL toolchain selected in `Languages & Frameworks | Rust` and then tries to run on a remote target, the WSL command will be executed despite remote being Linux.

This happens on my setup which is:
- Local machine: Windows 10
- Remote machine: Ubuntu 20.04

Running it with the WSL toolchain selected results in:

```
/home/mari/.cargo/bin/cargo --distribution Ubuntu --exec /bin/sh -c export PATH="/home/mari/.cargo/bin:$PATH" && export RUST_BACKTRACE=short && export TERM=ansi && cd /mnt/c/Users/Mari/Programming/<censored> && /home/mari/.cargo/bin/cargo run --color=always --target-dir=/home/mari/Programming/<censored>/target -- <censored>
error: Found argument '--distribution' which wasn't expected, or isn't valid in this context

	If you tried to supply `--distribution` as a value rather than a flag, use `-- --distribution`

USAGE:
    cargo [+toolchain] [OPTIONS] [SUBCOMMAND]

For more information try --help

Process finished with exit code 1
```

Selecting the normal Windows toolchain in `Languages & Frameworks | Rust` results in the correct command being ran:

```
/home/mari/.cargo/bin/cargo run --color=always --target-dir=/home/mari/Programming/<censored>/target -- <censored>
    Finished dev [unoptimized + debuginfo] target(s) in 0.20s
```

## Steps to reproduce

1. Setup Rust Plugin using WSL toolchain in `Languages & Frameworks | Rust`
2. Setup remote target as described here: https://www.jetbrains.com/help/clion/remote-projects-support.html
3. Run a configuration with the target being the remote machine like this:
![img](https://i.cutegirl.tech/m9bs2z4z9icq.png)

**Comment by chrispollaro** [comment]
I ran into the issue, but I can get it working as long as the toolchain is set as default.  However, I used the WSL instructions here: https://www.jetbrains.com/help/clion/how-to-use-wsl-development-environment-in-product.html.  If you don't try to set the toolchain as the default and try to create CMAKE profile for it, it will run the cargo command with the arguments for the WSL.exe, which is what causes the error as --distribution is being passed to cargo and not the wsl.exe.  You also may need to delete any previous configurations once you have changed the toolchain to be safe.

![image](https://user-images.githubusercontent.com/19241781/178124726-3f1f47dc-2b3a-4d02-989d-907742266317.png)

I also noticed your working directory is set to your Windows machine.

![image](https://user-images.githubusercontent.com/19241781/178124753-5b8415fe-a249-42c9-ac45-f6a68ca19b31.png)
